frontend/src/components/CommunicationLogs/ViewCallLogs.tsx

```diff
@@ -0,0 +1,490 @@
+    setSelectionID(id);
+  };
+
+  const handleClickOpenDelete = () => {
+    setOpenDelteAlert(true);
+  };
+
+  const handleCloseDelete = () => {
+    setOpenDelteAlert(false);
+  };
+
+  const getClientInput = async (event: React.ChangeEvent<{}>, value: string) => {
+    const clientResponse = await getAllClientAccount(value);
+    const clients: any[] = [];
+    clientResponse.data.forEach((element: any) => {
+      clients.push(`${element.email}`);
+    });
+    setClientList(clients);
+  };
+
+  const clearStates = () => {
+    setSelectedClient('');
+    setAddedNote('');
+    setSelectedIndexAction(1);
+  };
+
+  const handleSubmitLog = async () => {
+    const data: CallCreationDTO = {
+      receiverEmail: selectedClient as string,
+      description: addedNote as string,
+      action: actions[selectedIndexAction].toUpperCase(),
+      date: new Date().toISOString().split('T')[0],
+      followUp: false,
+      neverCallBack: false,
+      callerEmail: account.account.email,
+    };
+    try {
+      console.log(data);
+      await createCall(data);
+      setOpenDialog(false);
+      clearStates();
+      fetchData();
+    } catch (err: any) {
+      history.push('/error');
+    }
+  };
+
+  const handleUpdatedLog = async () => {
+    const data: CallUpdateDTO = {
+      receiverEmail: selectedClient as string,
+      description: addedNote as string,
+      action: actions[selectedIndexAction].toUpperCase(),
+      date: new Date().toISOString().split('T')[0],
+      followUp: false,
+      neverCallBack: false,
+      callerEmail: account.account.email,
+    };
+    try {
+      await updateCall(selectID as number, data);
+      setOpenDialog(false);
+      setEditState(false);
+      clearStates();
+      fetchData();
+    } catch (err: any) {
+      history.push('/error');
+    }
+  };
+
+  const handleDeleteLog = async () => {
+    try {
+      await deleteCall(selectID as number);
+      setOpenDelteAlert(false);
+      fetchData();
+    } catch (err: any) {
+      history.push('/error');
+    }
+  };
+
+  const fetchData = async () => {
+    const calls = await getCalls(account.account.email);
+    const display: TableData[] = [];
+    calls.data.forEach((element: Data) => {
+      const date = element.date.split('T');
+      const dataDisplay: TableData = {
+        id: element.id,
+        receiverName: element.receiverName,
+        phoneNumber: element.phoneNumber,
+        action: element.action.charAt(0) + element.action.substring(1).toLowerCase(),
+        date: date[0],
+        description: element.description,
+        receiverEmail: element.receiverEmail,
+        callerEmail: element.callerEmail,
+        followUp: element.followUp,
+        neverCallBack: element.neverCallBack,
+        time: date[1].substring(0, 8),
+      };
+      display.push(dataDisplay);
+    });
+    setCallsDisplayed(display);
+  };
+
+  useEffect(() => {
+    fetchData();
+    // eslint-disable-next-line
+  }, [select]);
```

Comment:
try adding fetchData to the array and remove the comment. It should work.

Reply:
Wouldn't that create an infinite loop as every time fetchData is called, useEffect might also be called.

Reply:
The way it is right now, it doesn't create an infinite loop